lib/pure/strutils.nim

```diff
@@ -649,22 +650,29 @@ iterator splitLines*(s: string): string =
   ##   ""
   var first = 0
   var last = 0
+  var eolpos = 0
   while true:
     while last < s.len and s[last] notin {'\c', '\l'}: inc(last)
-    yield substr(s, first, last-1)
-    # skip newlines:
-    if last >= s.len: break
+
+    if last >= s.len:
+      yield substr(s, first, last-1)
```

Comment:
These additional ``yield``'s can cause serious code bloat.

Reply:
I see, I'm not familiar with how Nim's generates the state machine to handle the generator state, in other languages I've worked with once you have a generator additional yields are not much of an issue, `except` blocks is usually what can significally increase the bloat.

I'll rework the logic to keep a single `yield` in the function.